src/main/java/com/commercetools/sync/types/utils/FieldDefinitionUpdateActionUtils.java

```diff
@@ -0,0 +1,110 @@
+package com.commercetools.sync.types.utils;
+
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.types.EnumFieldType;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.LocalizedEnumFieldType;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.commands.updateactions.ChangeFieldDefinitionLabel;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.types.utils.TypeUpdateLocalizedEnumActionUtils.buildLocalizedEnumValuesUpdateActions;
+import static com.commercetools.sync.types.utils.TypeUpdatePlainEnumActionUtils.buildEnumValuesUpdateActions;
+import static java.util.stream.Collectors.toList;
+
+
+public final class FieldDefinitionUpdateActionUtils {
+
+    /**
+     * Compares all the fields of an {@link FieldDefinition} and an {@link FieldDefinition} and returns
+     * a list of {@link UpdateAction}&lt;{@link Type}&gt; as a result. If both the {@link FieldDefinition}
+     * and the {@link FieldDefinition} have identical fields, then no update action is needed and hence an
+     * empty {@link List} is returned.
+     *
+     * @param oldFieldDefinition  the field definition which should be updated.
+     * @param newFieldDefinition the field definition draft where we get the new fields.
+     * @return A list with the update actions or an empty list if the field definition fields are identical.
+     */
+    @Nonnull
+    public static List<UpdateAction<Type>> buildActions(
+        @Nonnull final FieldDefinition oldFieldDefinition,
+        @Nonnull final FieldDefinition newFieldDefinition) {
+
+        final List<UpdateAction<Type>> updateActions;
+
+        updateActions = Stream
```

Comment:
I think I forgot this comment but the code does the same thing.

Reply:
What do you mean?

Reply:
Why this code is better?

Reply:
Because you don't have to create Stream from it. It doesn't have any meaning here and makes the code more complex. Rather use what Optional offers by default.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/304/commits/f7a4c768dd77bdb7892f0d9aa92cd604330a2aa7